core/src/main/scala/com/microsoft/azure/synapse/ml/causal/LinearDMLParams.scala

```diff
@@ -0,0 +1,109 @@
+import org.apache.spark.ml.regression.Regressor
+
+trait HasTreatmentCol extends Params {
+  val treatmentCol = new Param[String](this, "treatmentCol", "treatment column")
+  def getTreatmentCol: String = $(treatmentCol)
+
+  /**
+   * Set name of the column which will be used as treatment
+   *
+   * @group setParam
+   */
+  def setTreatmentCol(value: String): this.type = set(treatmentCol, value)
+}
+
+trait HasOutcomeCol extends Params {
+  val outcomeCol: Param[String] = new Param[String](this, "outcomeCol", "outcome column")
+  def getOutcomeCol: String = $(outcomeCol)
+
+  /**
+   * Set name of the column which will be used as outcome
+   *
+   * @group setParam
+   */
+  def setOutcomeCol(value: String): this.type = set(outcomeCol, value)
+}
+
+trait LinearDMLParams extends Params
+  with HasTreatmentCol with HasOutcomeCol with HasFeaturesCol
+  with HasMaxIter with HasWeightCol with HasParallelismInjected {
+
+  val treatmentModel = new EstimatorParam(this, "treatmentModel", "treatment model to run")
+  def getTreatmentModel: Estimator[_ <: Model[_]] = $(treatmentModel)
+
+  /**
+   * Set treatment model, it could be any model derived from 'org.apache.spark.ml.regression.Regressor' or 'org.apache.spark.ml.classification.ProbabilisticClassifier'
+   *
+   * @group setParam
+   */
+  def setTreatmentModel(value: Estimator[_ <: Model[_]]): this.type = {
+    val isSupportedModel = value match {
+      case regressor: Regressor[_,_,_] =>   true // for continuous treatment
+      case classifier: ProbabilisticClassifier[_, _, _] => true
+      case _ => false
+    }
+    if (!isSupportedModel)  {
+      throw new Exception("LinearDML only support regressor and ProbabilisticClassifier as treatment model")
+    }
+    set(treatmentModel, value)
+  }
+
+  val outcomeModel = new EstimatorParam(this, "outcomeModel", "outcome model to run")
+  def getOutcomeModel: Estimator[_ <: Model[_]] = $(outcomeModel)
+
+  /**
+   * Set outcome model, it could be any model derived from 'org.apache.spark.ml.regression.Regressor' or 'org.apache.spark.ml.classification.ProbabilisticClassifier'
+   *
+   * @group setParam
+   */
+  def setOutcomeModel(value: Estimator[_ <: Model[_]]): this.type = {
+    val isSupportedModel = value match {
+      case regressor: Regressor[_,_,_] =>   true
+      case classifier: ProbabilisticClassifier[_, _, _] => true
+      case _ => false
+    }
+    if (!isSupportedModel)  {
+      throw new Exception("LinearDML only support regressor and ProbabilisticClassifier as outcome model")
+    }
+    set(outcomeModel, value)
+  }
+
+  val sampleSplitRatio = new UntypedArrayParam(this,
+    "SampleSplitRatio", "SampleSplitRatio")
+  def getSampleSplitRatio: Array[Double] = $(sampleSplitRatio).map(_.toString.toDouble)
+
+  /**
+   * Set the sample split ratio, default is Array(0.5, 0.5)
+   *
+   * @group setParam
+   */
+  def setSampleSplitRatio(value: Array[Any]): this.type = set(sampleSplitRatio, value)
```

Comment:
```suggestion
  val sampleSplitRatio = new DoubleArrayParam(
    this,
    "SampleSplitRatio",
    "Sample split ratio for cross-fitting. Default: [0.5, 0.5].",
    split => split.length == 2 && split.forall(_ >= 0)
  )

  def getSampleSplitRatio: Array[Double] = $(sampleSplitRatio).map(v => v / $(sampleSplitRatio).sum)

  /**
   * Set the sample split ratio, default is Array(0.5, 0.5)
   *
   * @group setParam
   */
  def setSampleSplitRatio(value: Array[Double]): this.type = set(sampleSplitRatio, value)
```

Reply:
There is no DoubleArrayParam, that's why I made the code a little weird. Let me try add DoubleArrayParam

Reply:
It's spark's param, thought I should keep using Synapse's version.